docs/DeveloperGuide.md

```diff
@@ -1,34 +1,315 @@
+  
+    Main Menu:
+    [1] Module Information
+    [2] CAP Simulator/Calculator
+    [3] Task Manager
+    [4] External Links
+    [5] Exit Program
+    ````
+
+---
+
+## 3. Design
+
+### 3.1 Architecture
+
+The `Duke` class contains the main method of our application.
+
+- At the start of the application, it initializes all the classes in sequence.
+
+We have 5 main classes for our key features.
+
+- `ModuleInfo`: contains methods for our module information features.
+- `HelpGraduation`: contains methods for our SU calculator/simulator.
+- `TaskManager`: contains methods for our task management features.
+- `Links`: contains methods for our links features.
+- `Storage`: Write data to, and loads data to hard disk
+
+![Diagram](diagrams/MainArchitecture.png)
+
+### 3.2 UI Class
+
+This class handles the interactions with the user and contains the methods for system output.
+
+---
+
+## 4. Implementation
+
+### 4.01 Add New Review
+
+This feature allows user to `add a review` for any of the modules they have taken.
+
+**NOTE**: This action will **overwrite** any existing reviews with the new review.
+
+E.g. Input:  
+`CS2113T is very fun:) I have learnt so many important concepts.`  
+`Overall 10/10 :)`  
+`/end`
+
+Output:
+
+```
+Woohoo~ Review added:  
+CS2113T is very fun:) I have learnt so many important concepts.  
+Overall 10/10 :) 
+
+Returning to module information menu...
+  ```
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewReview.png)  
+A general explanation of how this feature works:
+
+**Problem**: When the user wants to review a module, he/she should be able to type ___multiple lines
+of input___, as reviews are generally long and detailed.
+
+**Solution**: A `while` loop is used to take in multiple lines of input until the input contains
+the `/end` symbol. Anything typed after this `/end` symbol will be erased.
+
+**When overwriting this review with a new review**: The old review will be printed for user
+reference. The user will also be alerted that this action will delete his/her old review via a log.
+The log is printed out as a warning.
+
+
+### 4.02 Delete a Review
+
+This feature allows user to delete a review from the `module`.
+
+A general explanation of how this feature works:
+
+Deleting a review would __reset__ the `Review` *String* attribute of a `module` to a default value
+of:  
+`"You have not reviewed this module yet."`
+
+### 4.03 Add New Task
+
+This feature allows the user to add tasks of type `task`, `assignment`, `midterm`, `final exam` to a
+task list.
+
+All task types (except `Task`, which contains arguments 1, 2 and 5) contain 5 arguments upon
+creation:
+
+1) `Module Code` <br> Example: `CS2113T`
+2) `Task Description` <br> Example: `Finish v2.0`
+3) `Date` <br> Example: `2021-03-21`
+4) `Time` <br> Example: `23:59`
+5) `Message` <br> Example: `Good job!`
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewTask.png)
```

Comment:
The sequence diagram is very detailed, as such it may be a bit lengthy and confusing? Consider splitting the blocks into separate diagrams and explaining them separately.